database/sql/11-functions-triggers.sql

```diff
@@ -78,4 +78,44 @@ $$ LANGUAGE plpgsql;
 CREATE TRIGGER notify_acquirablefile_geoprocess
 AFTER INSERT ON public.acquirablefile
 FOR EACH ROW
-EXECUTE PROCEDURE public.notify_acquirablefile_geoprocess();
\ No newline at end of file
+EXECUTE PROCEDURE public.notify_acquirablefile_geoprocess();
+
+--------------------------------------------------------------
+-- ASYNC LISTENER FUNCTION (ALF) FOR snodas_interpolate_geoprocess
+--------------------------------------------------------------
+
+-- Trigger Function; Inserts Into acquirablefile Table
+CREATE OR REPLACE FUNCTION public.notify_snodas_interpolate_geoprocess() RETURNS trigger AS $$
+    BEGIN
+        PERFORM (
+            WITH geoprocess_config as (
+                SELECT id                             AS productfile_id,
+                       product_id                     AS product_id,
```

Comment:
Recommend reducing the payload to the bare minimum required by p_snodas_interpolate.py.  Providing additional payload if not required makes it difficult in the future to know what code is required and what can be refactored or deleted.  I think the only fields required in `geoprocess_config` at present are `datetime` and `max_distance`.  Recommend we do add the payload value `write_to_bucket` and replace `CONFIG.WRITE_TO_BUCKET` with what is provided via payload in these locations:

- https://github.com/USACE/cumulus-api/blob/develop/async_geoprocess/geoprocess_worker/p_snodas_interpolate.py#L38
- https://github.com/USACE/cumulus-api/blob/develop/async_geoprocess/geoprocess_worker/p_snodas_interpolate.py#L60
- https://github.com/USACE/cumulus-api/blob/develop/async_geoprocess/geoprocess_worker/p_snodas_interpolate.py#L83
- https://github.com/USACE/cumulus-api/blob/develop/async_geoprocess/geoprocess_worker/p_snodas_interpolate.py#L107
- https://github.com/USACE/cumulus-api/blob/develop/async_geoprocess/geoprocess_worker/p_snodas_interpolate.py#L130

Reply:
totally agree.  changed have been made.